File: examples/zig/add-data-output/DataOutputTest_test.zig
const std = @import("std");
const root = @import("../examples_test.zig");

// DataOutputTest.runar.zig calls self.addDataOutput — a Rúnar intrinsic that
// the compiler materialises into the emitted Bitcoin Script, not a real Zig
// method on the contract struct. So we only exercise the Rúnar frontend
// (parse → validate → typecheck) via compileCheck.

fn contractPath(comptime basename: []const u8) []const u8 {
    return "add-data-output/" ++ basename;
}

test "compile-check DataOutputTest.runar.zig (source)" {
    try root.runar.compileCheckSource(
        std.testing.allocator,
        @embedFile("DataOutputTest.runar.zig"),
        "DataOutputTest.runar.zig",
    );
}

test "compile-check DataOutputTest.runar.zig (file)" {
    try root.runar.compileCheckFile(
        std.testing.allocator,
        contractPath("DataOutputTest.runar.zig"),
    );
}

File: examples/go/add-data-output/DataOutputTest_test.go
package contract

import (
	"testing"

	runar "github.com/icellan/runar/packages/runar-go"
)

// DataOutputTest.runar.go uses self.AddDataOutput, a Rúnar intrinsic the
// compiler materialises into emitted Bitcoin Script. Although the Go SDK's
// StatefulSmartContract base struct does provide an AddDataOutput method we
// could call natively, the cross-compiler conformance boundary we care about
// is the Rúnar frontend: parse → validate → typecheck. We exercise that
// directly via CompileCheck.

func TestDataOutputTest_Compile(t *testing.T) {
	if err := runar.CompileCheck("DataOutputTest.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}
